### DISCUSSION TITLE
Plot np.array as mesh

### DISCUSSION THREAD

**Discussion by SantiagoPendon** [pyvista/pyvista#5341] [Q&A] [answered]
Hi to everyone. My problem is the following one:

I read a binary np volume ( i'm 100% sure that is not empty) and then transform it as ImageData, to add it as a mesh. The procedure is:

```python
mask  = np.load(...)
grid = pv.ImageData()
p = pv.Plotter()
grid = pv.wrap(mask)
p.add_mesh(grid, opacity='sigmoid')
```

The mesh represented is  composed only by zero values, when the mask has zero and one values.

Any idea about what's going on?

**Comment by MatthewFlamm** [comment, marked as answer]
Can you share a minimum reproducible example with example data?  Your code looks like it is probably missing smoe information.  Is the `mask` just an array of 1's and 0's that is the same as a number of points?  If so, where is the point location data?

An example that works like I'd expect:

```py
import pyvista as pv
import numpy as np

points = np.array([[0., 0., 0.], [1., 0., 0.]])
mask = [1., 0.]
mesh = pv.wrap(points)
mesh["mask"] = mask

mesh.plot(opacity="sigmoid")
```

**Reply by SantiagoPendon** [reply]
I apologice for my inconvenient explanation. The truth is the np.volume named `mask`  is a 134x112x154 binary volume. What I'm trying to do is create a visualization of the 1's with a solid color. 

Something like this:

![Captura de pantalla 2023-12-15 084029](https://github.com/pyvista/pyvista/assets/151015554/7b59a69c-f713-43ee-8c48-f6bc9fe5ff33)


Can I do a mesh representation from the binary tensor based on i,j,k positions and value in each ijk?

![Captura de pantalla 2023-12-15 090405](https://github.com/pyvista/pyvista/assets/151015554/d26ec73f-6234-49ce-9055-9b18698767b9)


Thanks a lot Matthew!

**Reply by tkoyama010** [reply]
This is an excellent goal.

I guess that @MatthewFlamm wants you to share your data with the script you are using. It is good if the script and the data is simple, but even if it is not, script and data sharing can help us to know your situation.
